go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -266,9 +298,29 @@ func (c *Controller) syncService(key string) error {
 			},
 		}
 
-		if err := ovnlb.EnsureLBs(c.nbClient, service, nil); err != nil {
-			return fmt.Errorf("failed to delete load balancers for service %s/%s: %w",
-				namespace, name, err)
+		var existingLBs []LB
+		c.alreadyAppliedLock.Lock()
+		if alreadyAppliedLbs, ok := c.alreadyApplied[key]; ok {
+			existingLBs = make([]LB, len(alreadyAppliedLbs))
+			copy(existingLBs, alreadyAppliedLbs)
+		}
+		c.alreadyAppliedLock.Unlock()
+
+		if existingLBs != nil {
```

Comment:
although I doubt this would ever have a real issue, this is not the same as the previous logic. Fro example you could have:
```
serviceApplied["test"]= nil slice
```

Then the if statement would not be entered and the key would never get deleted from the map. It looks like this is never possible because when lbs are added to the cache they are never nil:

https://github.com/flavio-fernandes/ovn-kubernetes/blob/ensurelbs/go-controller/pkg/ovn/controller/services/services_controller.go#L360

So I think I'm being too paranoid. wdyt?

Reply:
never too paranoid. I want to fix it. TY for pointing this out.